lib/valueflow.cpp

```diff
@@ -1079,6 +1071,38 @@ static void valueFlowArray(TokenList *tokenlist)
     }
 }
 
+static void valueFlowArrayBool(TokenList *tokenlist)
+{
+    for (Token *tok = tokenlist->front(); tok; tok = tok->next()) {
+        if (tok->hasKnownIntValue())
+            continue;
+        const Variable *var = nullptr;
+        bool known = false;
+        std::list<ValueFlow::Value>::const_iterator val =
+            std::find_if(tok->values().begin(), tok->values().end(), std::mem_fn(&ValueFlow::Value::isTokValue));
+        if (val == tok->values().end()) {
+            var = tok->variable();
+            known = true;
+        } else {
+            var = val->tokvalue->variable();
+            known = val->isKnown();
+        }
+        if (!var)
+            continue;
+        if (!var->isArray() || var->isArgument() || var->isStlType())
+            continue;
+        // TODO: Check for function argument
+        if ((tok->valueType() && tok->valueType()->isIntegral()) ||
```

Comment:
if the pointer is converted to some plain integer.. then there is no pointer to bool conversion. The value will not be 1 but non-zero. Maybe you know that this will work fine but I feel I need to think a little about it.

Reply:
> The value will not be 1 but non-zero. 

We have no way of expressing that in ValueFlow, currently. We should consider adding a "impossible" type to ValueFlow in the future.